state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -1858,26 +1859,37 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 			Self::accrue_withheld_fee(asset, sp_std::cmp::min(fee_estimate, available_amount));
 			available_amount.saturating_sub(fee_estimate)
 		} else {
+			// In case of an none-gas asset, we need to use the simulation amount. If the simulation
+			// amount is not available, we use the available amount.
+			let simulation_amount = if let Some(simulation_amount) =
+				get_simulation_amount(asset.into())
+			{
+				simulation_amount
+			} else {
+				log::warn!("No simulation amount available for {asset:?}. Using available amount.");
+				available_amount.into()
```

Comment:
We could avoid this branch and potential error case if we update the asset macro, adding a `NonNativeAsset` type for example (not sure how simple this would be). 

I think it's worth trying. Leaving it as it is, it's easy to forget to add a variant to `get_simulation_amount` if we add a new asset. And then we need a runtime upgrade to fix it. 

I would suggest we either add the new type as suggested above (-> compile time guarantees) OR if that proves impractical, add the simulation amounts as storage items with a governance update extrinsics (-> flexibility).